src/sql/src/statements.rs

```diff
@@ -172,7 +172,15 @@ macro_rules! parse_number_to_value {
                     }.fail()
                 }
             }
-
+            // It's valid for MySQL JDBC to send "0" and "1" for boolean types, so adapt to that.
+            ConcreteDataType::Boolean(_) => {
+                match $n {
+                    "0" => Ok(Value::Boolean(false)),
+                    "1" => Ok(Value::Boolean(true)),
+                    _ => ParseSqlValueSnafu {
+                        msg: format!("Failed to parse number '{}' to boolean column type!", $n)}.fail(),
```

Comment:
```suggestion
                        msg: format!("Failed to parse number '{}' to boolean column type", $n)}.fail(),
```

nit: we may not use this tone else where.